File: Server/src/main/java/org/example/entities/ProductEntity.java
package org.example.entities;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@RequiredArgsConstructor
@AllArgsConstructor
@Data
@Entity
@Table(name = "tbl_products")
public class ProductEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", length = 250, nullable = false)
    private String name;

    @Column(name = "price", nullable = false)
    private Double price;

    @Column(name = "description", length = 5000, nullable = true)
    private String description;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<ProductImageEntity> productImages;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private CategoryEntity category;
}

File: Server/src/main/java/org/example/entities/ProductEntity.kt
package org.example.entities

import jakarta.persistence.*
import lombok.RequiredArgsConstructor

@RequiredArgsConstructor
@Entity
@Table(name = "tbl_products")
class ProductEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    lateinit var name: String

    var price: Double? = null

    lateinit var description: String

    @OneToMany(mappedBy = "product", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    lateinit var productImages: List<ProductImageEntity>

    @ManyToOne(fetch = FetchType.LAZY)
    lateinit var category: CategoryEntity
}
